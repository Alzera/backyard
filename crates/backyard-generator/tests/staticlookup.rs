use backyard_generator::generate;
use backyard_parser::parse_eval;

#[test]
fn basic() {
  let asts = parse_eval("self::a;").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn bracket() {
  let asts = parse_eval("self::{$a};").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn test_class() {
  let asts = parse_eval("self::class;").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}

#[test]
fn variable() {
  let asts = parse_eval("self::$a;").unwrap();
  insta::assert_yaml_snapshot!(generate(&asts).unwrap());
}
