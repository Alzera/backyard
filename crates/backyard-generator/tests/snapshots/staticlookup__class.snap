---
source: crates/backyard-generator/tests/staticlookup.rs
expression: generate(asts).unwrap()
snapshot_kind: text
---
"self::class;"
